xarray/backends/h5netcdf_.py

```diff
@@ -468,7 +468,7 @@ def guess_can_open(self, filename_or_obj: T_PathFileOrDataStore) -> bool:
             return magic_number.startswith(b"\211HDF\r\n\032\n")
 
         if isinstance(filename_or_obj, str | os.PathLike):
-            _, ext = os.path.splitext(filename_or_obj)
+            _, ext = os.path.splitext(str(filename_or_obj))
```

Comment:
Does this need casting with `str`? the docs for `os.path.splitext` note that it accepts path-like objects